hoot-core-test/src/test/cpp/hoot/core/visitors/WayGeneralizeVisitorTest.cpp

```diff
@@ -63,15 +66,65 @@ class WayGeneralizeVisitorTest : public HootTestFixture
       map,
       "test-files/visitors/RandomWayGeneralizerTest/RandomWayGeneralizerTest-in-1.osm");
 
-    WayGeneralizeVisitor wayGeneralizeVisitor;
-    wayGeneralizeVisitor.setEpsilon(5.0);
-    map->visitRw(wayGeneralizeVisitor);
+    WayGeneralizeVisitor uut;
+    uut.setEpsilon(5.0);
+    uut.setRemoveNodesSharedByWays(true);
+    map->visitRw(uut);
 
     const QString outputFile = _outputPath + "runBasicTest.osm";
     OsmMapWriterFactory::write(map, outputFile);
 
     HOOT_FILE_EQUALS(_inputPath + "runBasicTest.osm", outputFile);
   }
+
+  void runCritTest()
+  {
+    OsmMapPtr map(new OsmMap());
+    OsmMapReaderFactory::read(
+      map, "test-files/conflate/unified/AllDataTypesA.osm", false, Status::Unknown1);
+
+    WayGeneralizeVisitor uut;
+    uut.setEpsilon(0.1);
+    uut.setRemoveNodesSharedByWays(true);
+    uut.addCriterion(std::shared_ptr<HighwayCriterion>(new HighwayCriterion()));
+    map->visitRw(uut);
+
+    const QString outputFile = _outputPath + "runCritTest.osm";
+    OsmMapWriterFactory::write(map, outputFile);
+
+    HOOT_FILE_EQUALS(_inputPath + "runCritTest.osm", outputFile);
+  }
+
+  void runRemoveSharedNodeTest()
```

Comment:
Test to make sure you don't remove a way node during generalization that's shared by multiple ways.